### PR TITLE
gateway: fix timing issue in unittest

### LINKED ISSUE #449: unittest failed
I don't understand how  `inbound` could get more than one discovery-request...  

```
[ RUN      ] gateway_manager.A_B__pending_call_A_to_B__shutdown_B__send_implicit_update_to_A_outbound__expect_direct_reply__not_over_to_inbound
/git/casual/casual/middleware/gateway/unittest/source/test_manager.cpp:1541: Failure
Value of: found->received == 1
  Actual: false
Expected: true
*found: { type: domain_discovery_request, sent: 2, received: 2}
/git/casual/casual/middleware/gateway/unittest/source/test_manager.cpp:1541: Failure
Value of: found->received == 1
  Actual: false
Expected: true
*found: { type: domain_discovery_request, sent: 2, received: 2}
/git/casual/casual/middleware/gateway/unittest/source/test_manager.cpp:1541: Failure
Value of: found->received == 1
  Actual: false
Expected: true
*found: { type: domain_discovery_request, sent: 2, received: 2}
/git/casual/casual/middleware/gateway/unittest/source/test_manager.cpp:1541: Failure
Value of: found->received == 1
  Actual: false
Expected: true
*found: { type: domain_discovery_request, sent: 2, received: 2}
/git/casual/casual/middleware/gateway/unittest/source/test_manager.cpp:1541: Failure
Value of: found->received == 1
  Actual: false
Expected: true
*found: { type: domain_discovery_request, sent: 2, received: 2}
[  FAILED  ] gateway_manager.A_B__pending_call_A_to_B__shutdown_B__send_implicit_update_to_A_outbound__expect_direct_reply__not_over_to_inbound (334 ms)
```

### PR DESCRIPTION
**Pull request by AlbsAgren**
Before: One of the gateway unittests had an issue where the accumulate mechanism in casual-domain-discovery could cause an additional discovery to be sent, contrary to the assumptions of the test.

Now: We wait for the accumulate to time out, ensuring that only the service call will trigger a discovery.

Resolves #449

### CODE REVIEW

File: middleware/gateway/unittest/source/test_manager.cpp [hunk 1]
```diff
@@ -1487,6 +1487,25 @@ namespace casual
...
+                  common::message::counter::Request{ common::process::handle()});
+            });
+
+            auto received_fetch_known = []( auto& reply)
+            {
+               auto found = algorithm::find( reply.entries, common::message::Type::domain_discovery_fetch_known_request);
+               return found && found->received == 1;
+            };
+
+            fetch_sm_message_counter_until( received_fetch_known);
```
Comment (lazan) [inline middleware/gateway/unittest/source/test_manager.cpp]: This might be a good thing to extract to common unittest utility?

**Review by lazan — APPROVED** [review]
Looking good.

**Review by ahhud — APPROVED** [review]

File: middleware/common/include/common/unittest.h [hunk 2]
```diff
@@ -184,6 +184,21 @@ namespace casual
...
+            auto until( D&& device, F predicate)
+            {
+               auto fetcher = fetch::until( [ &device]
+               {
+                  return communication::ipc::call( std::forward< D>( device), common::message::counter::Request{ common::process::handle()});
+               });
+
+               return fetcher( predicate);
+            }
+         } // message::counter
```
Comment (lazan) [inline middleware/common/include/common/unittest.h:200]: Nice

**Review by lazan — APPROVED** [review]
Looking good

**Pull request merged**